java/examples/src/main/java/com/nedap/retail/services/examples/EpcExample.java

```diff
@@ -1,32 +1,377 @@
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.joda.time.DateTime;
+
 import com.nedap.retail.messages.Client;
 import com.nedap.retail.messages.ClientException;
+import com.nedap.retail.messages.article.Article;
+import com.nedap.retail.messages.article.Barcode;
+import com.nedap.retail.messages.article.Size;
+import com.nedap.retail.messages.epc.v2.approved_difference_list.ApprovedDifferenceListSummary;
+import com.nedap.retail.messages.epc.v2.approved_difference_list.request.ApprovedDifferenceListCaptureRequest;
+import com.nedap.retail.messages.epc.v2.approved_difference_list.response.ApprovedDifferenceListExportResponse;
+import com.nedap.retail.messages.epc.v2.approved_difference_list.response.ApprovedDifferenceListResponse;
 import com.nedap.retail.messages.epc.v2.difference_list.DifferenceListResponse;
+import com.nedap.retail.messages.epc.v2.stock.NotOnShelfRequest;
+import com.nedap.retail.messages.epc.v2.stock.NotOnShelfResponse;
 import com.nedap.retail.messages.epc.v2.stock.StockResponse;
+import com.nedap.retail.messages.organization.Location;
+import com.nedap.retail.messages.organization.LocationSubType;
+import com.nedap.retail.messages.stock.GtinQuantity;
+import com.nedap.retail.messages.stock.Stock;
 
 public class EpcExample {
 
-    private static final String LOCATION = "http://retailer.com/loc/123";
-    private static final String ERP_STOCK_ID = "72c49313dc6c0f55500974fbe6968606cd0076bdf004884dbb4b6a85291712f5";
+    private static final DateTime EVENT_TIME = DateTime.now();
+    private static final String EXTERNAL_REFERENCE = "generated_import";
+    private static final String NEW_LINE = System.lineSeparator();
+    private static final String TAB = "\t";
+    private static final String DOUBLE_TAB = "\t\t";
+    private static final String GTIN_1 = "02011200000019";
+    private static final String GTIN_2 = "02011200000064";
+    private static final String GTIN_3 = "02011200000163";
+    private static final String COMMA = ", ";
+    private static final String WHITESPACE = " ";
+    private static final String COLON = ":";
 
     public static void runExample(final Client client) {
-        System.out.println("*** EPC API example ***");
+        System.out.println(NEW_LINE + "*** EPC API example ***");
+
+        final String locationId = client.getSites().get(0).getId();
 
         try {
-            // get difference list
-            System.out.println("Retrieving difference list...");
-            final DifferenceListResponse dl = client.differenceList(ERP_STOCK_ID, null, null, null);
-            System.out.println(dl.toString());
+            // Create some articles
+            System.out.println(NEW_LINE + "Creating some articles...");
+            client.captureArticles(Arrays.asList(createArticle1(), createArticle2(), createArticle3()));
```

Comment:
Yes, that is a great idea.
Creating example data directly from the example client would be really helpfully (not only to system integrators, but also for us as developers).

But lately, I have been thinking about the "generate..." endpoints:
Instead of calling "generate..." endpoints, we could also provide the library for generating items as open source (i.e. include in this project).
- When it is included in this project, we can directly create example data (without calling "generate..." endpoints.
- It can also serve as documentation for system integrators.
- The "generate..." endpoints we use the open source library for generating data.
- It's also possible for blackbox test to use the open source library for generating data.